packages/loader/driver-definitions/src/storage.ts

```diff
@@ -278,8 +278,12 @@ export interface IDocumentService {
      * Useful for storage to commit any pending state if any (including any local caching).
      * Please note that it does not remove the need for caller to close all active delta connections,
      * as storage may not be tracking such objects.
+     * @param criticalError - tells if container (and storage) are closed due to critical error.
+     * Error might be due to disconnect between client & server knowlege about file, like file being overwritten
+     * in storage, but client having stale local cache.
+     * If driver implements any kind of local caching, such caches needs to be cleared on on critical errors.
      */
-    dispose(): void;
+    dispose(criticalError: boolean): void;
```

Comment:
dispose already has an overload to take an error. i don't really like this non-standard overload: https://github.com/microsoft/FluidFramework/blob/02bd5378e75ab0c28b13f63e702e740365fda321/common/lib/common-definitions/src/disposable.ts#L8

Reply:
this doesn't implement the dispose interface, which is also a bit strange

Reply:
i think i perfer we pass an error rather than a boolean, and let the driver handle the error how it wants

Reply:
sure, will use more generic version.